pkl-core/src/test/files/LanguageSnippetTests/output/api/string.pcf

```diff
@@ -109,6 +109,14 @@ examples {
     List("aa", "caa", "c")
     List("aabb", "aabb")
   }
+  ["splitLimit()"] {
+    List("aa")
+    List("aa", "caa")
+    List("aa", "caa", "caa", "caa")
+    List("aabbccaabbccaabbccaabbcc")
+    List("aabb", "aabbccaabbccaabbcc")
+    List("aabb", "aabb", "aabb", "aabbcc")
```

Comment:
Separate comment (for visibility, because former is on `Outdated` code), but also to put in a `suggestion`;

I played with Java's `String::split`, GoLang's `strings.SplitN`, and Python's `str.split`. For each of them, the following invariant holds:

```
let (result = string.splitLimit(separator, limit))
  string.endsWith(separator) | string.endsWidth(result.last)
```

Another invariant diverges between, on the one hand, Java & GoLang and, on the other, Python. 
```
string.splitLimit(separator, limit).length.isBetween(0, if (isPython) limit + 1 else limit)
```
(i.e. Java & GoLang count maximum post-split chunks, Python counts maximum separator occurrences)

Here, I was confused by the lack of the first invariant I mentioned, but upon closer inspection, that's only for the `Regex` case; your `String` case behaves like I would expect.
```suggestion
    List("aabbccaabbccaabbccaabbcc")
    List("aa", "caabbccaabbccaabbcc")
    List("aa", "caa", "caa", "caabbcc")
    List("aabbccaabbccaabbccaabbcc")
    List("aabb", "aabbccaabbccaabbcc")
    List("aabb", "aabb", "aabb", "aabbcc")
```

(Can you tell I would really _love_ property-based testing for Pkl?)